### PR TITLE
Use Closure instead of callable in MiddlewareQueue

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/Http/MiddlewareQueue.php [hunk 1]
```diff
@@ -92,13 +92,13 @@ protected function resolve($middleware): MiddlewareInterface
             return new DoublePassDecoratorMiddleware($middleware);
         }
 
-        return new CallableDecoratorMiddleware($middleware);
+        return new ClosureDecoratorMiddleware($middleware);
```
Comment (markstory) [inline src/Http/MiddlewareQueue.php]: If we wanted to maximize compatibility we could leave the accepted types of `add()` and `prepend()` as they are today, and use `Closure::fromCallable()` here when the provided type isn't a Closure.

Comment (ADmad) [reply src/Http/MiddlewareQueue.php]: This class has always handled closures only, the class naming and typehints were an oversight on my part. Any other callable is assumed to be 3.x style double pass middleware and is handled by the `DoublePassDecoratorMiddleware` class.

**Comment by markstory** [comment]
Looks good to me.

**Comment by ADmad** [comment]
Should we throw a warning for old style double pass middlewares and nudge people towards updating to new PSR15 middlewares?

**Pull request merged**